pkg/controller/controller_test.go

```diff
@@ -1813,7 +1953,95 @@ func assertServiceInstanceCredentialRequestRetriableError(t *testing.T, obj runt
 	assertServiceInstanceCredentialReadyCondition(t, obj, readyStatus, reason)
 	assertServiceInstanceCredentialCurrentOperation(t, obj, operation)
 	assertServiceInstanceCredentialOperationStartTimeSet(t, obj, true)
-	assertServiceInstanceCredentialReconciliationNotComplete(t, obj)
+	assertServiceInstanceCredentialReconciledGeneration(t, obj, originalBinding.Status.ReconciledGeneration)
+	switch operation {
+	case v1alpha1.ServiceInstanceCredentialOperationBind:
+		assertServiceInstanceCredentialInProgressPropertiesParameters(t, obj, inProgressParameters, inProgressParametersChecksum)
+	case v1alpha1.ServiceInstanceCredentialOperationUnbind:
+		assertServiceInstanceCredentialInProgressPropertiesNil(t, obj)
+	}
+	assertServiceInstanceCredentialExternalPropertiesUnchanged(t, obj, originalBinding)
+}
+
+func assertServiceInstanceCredentialInProgressPropertiesNil(t *testing.T, obj runtime.Object) {
```

Comment:
I think it would be good to have explanatory godoc for these methods but I do not want to block this PR on it.  Will you promise to do it in a follow-up?

Reply:
I promise. I will create an issue for it to remind myself.